pkg/util/secret.go

```diff
@@ -15,10 +15,15 @@ import (
 
 // GetTargetSecret will get secrets(type=targetType, namespace=targetNamespace) from a list of secret references.
 func GetTargetSecret(client kubeclient.Interface, secretReferences []corev1.ObjectReference, targetType corev1.SecretType, targetNamespace string) (*corev1.Secret, error) {
+	var errNotFound error
```

Comment:
Why do we need it?
I think the previous version would be fine.

Reply:
https://github.com/karmada-io/karmada/blob/193e00d832e34a2b1521044d3fb3afe45a610efa/pkg/karmadactl/join.go#L269-L275

Compatible with the check on L272